numpy/core/function_base.py

```diff
@@ -123,25 +123,47 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
     if num < 0:
         raise ValueError("Number of samples, %s, must be non-negative." % num)
     div = (num - 1) if endpoint else num
+    # Make sure one can use variables that have an __array_interface__, gh-6634
+    start = asanyarray(start)
+    stop  = asanyarray(stop)
+
+    # check if dealing with datetime64 or timedelta64
+    is_datelike = start.dtype.kind in 'Mm'
+    if is_datelike:
+        # For datetime/timedelta, dtype must be specified
+        # start and stop will first be cast to this dtype.
+        # Warning: dtype must have high enough precision so
+        # that it is at all possible to draw num equispaced
+        # samples.
+        assert dtype is not None, "linspace with {} objects cannot be called "\
+            "with dtype=None".format(start.dtype)
```

Comment:
Asserts should not be used for checks that users can run into. This should be a ValueError probably. I could imagine writing it without negation may be a tiny bit nicer.

Reply:
Thank you for this comment. I have changed this assert into a `ValueError`.